assembl/static2/js/app/components/common/media.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { I18n } from 'react-redux-i18n';
+import { Grid, Row, Col } from 'react-bootstrap';
+
+const isValidDescription = (description) => {
+  return !!(description && description !== '<p></p>');
+};
+
+const createHTMLTag = (contentType, url) => {
+  return (
+    (contentType && { image: <img src={url} alt="media" /> }[contentType.split('/')[0]]) ||
+    <object data={url} aria-label="media" />
+  );
+};
+
+const fetchContentType = (url) => {
+  return fetch(url, {
+    method: 'HEAD'
+  }).then(
+    (response) => {
+      return response.headers.get('Content-Type');
+    },
+    () => {
+      return null;
+    }
+  );
+};
+
+class Media extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { contentType: null };
+  }
+  componentWillMount() {
+    fetchContentType(this.props.htmlCode).then(this.setContentType);
```

Comment:
Should we create an action/reducer to do that?

Reply:
I don't think so since the component is used in multiple places (home, thread and survey)

Reply:
(not in home, my bad)